steward/src/cork.rs

```diff
@@ -94,6 +108,121 @@ impl steward::contract_call_server::ContractCall for CorkHandler {
+                    Function::EnterPosition(_) => contract_call(contract.enter_position()).await,
+                    Function::SetDepositLimit(params) => {
+                        contract_call(contract.set_deposit_limit(params.limit.into())).await
+                    }
+                    Function::Reinvest(params) => {
+                        contract_call(contract.reinvest(params.min_assets_out.into())).await
+                    }
+                    Function::Rebalance(params) => {
+                        contract_call(
+                            contract.rebalance(
+                                params
+                                    .route
+                                    .iter()
+                                    .map(|addr| match addr.parse::<H160>() {
+                                        Ok(addr) => Ok(addr),
+                                        Err(_) => Err(addr),
+                                    })
+                                    .collect::<Vec<_>>()
+                                    .iter()
+                                    .map(|r| r.unwrap())
+                                    .collect::<Vec<H160>>()
+                                    .try_into()
+                                    .expect("failed to convert 'route' addresses to array"),
+                                params
+                                    .swap_params
+                                    .iter()
+                                    .map(|sp| {
+                                        let out: [U256; 3] = [
+                                            sp.in_index.into(),
+                                            sp.out_index.into(),
+                                            sp.swap_type.into(),
+                                        ];
+                                        out
+                                    })
+                                    .collect::<Vec<[U256; 3]>>()
+                                    .try_into()
+                                    .expect("failed to convert 'swap_params' vec to array"),
+                                params.min_assets_out.into(),
+                            ),
+                        )
+                        .await
+                    }
+                    Function::AccrueFees(_) => contract_call(contract.accrue_fees()).await,
+                    Function::TransferFees(_) => contract_call(contract.transfer_fees()).await,
+                    Function::SetLiquidityLimit(params) => {
+                        contract_call(contract.set_liquidity_limit(params.limit.into())).await
+                    }
+                    Function::ClaimAndUnstake(_) => {
+                        contract_call(contract.claim_and_unstake()).await
+                    }
+                }
+                .unwrap_or_else(|err| {
+                    panic!("{:?}", tonic::Status::invalid_argument(err.to_string()));
```

Comment:
Using `panic!` here isn't helpful, we should be returning some kind of error back via the `Result` type of the `submit` function we're defining here. There are different `tonic` error types one could choose, here is an example of one: https://github.com/PeggyJV/steward/blob/1d761433f806ed9dbf4439432a04ca95590d4b55/steward/src/cork.rs#L86-L89

Reply:
Ooh, I thought we could just panic and display the `tonic::Status::invalid_argument` tonic error type.

Reply:
done!